src/test/java/org/jenkinsci/plugins/gitclient/GitUsernamePasswordBindTest.java

```diff
@@ -0,0 +1,81 @@
+package org.jenkinsci.plugins.gitclient;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class GitUsernamePasswordBindTest {
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {"randomName", "special%%_342@**"}
+        });
+    }
+
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+
+
+    private final String username;
+
+    private final String password;
+
+    public GitUsernamePasswordBindTest(String username, String password){
+        this.username = username;
+        this.password = password;
+    }
+
+    private File rootDir = null;
+    private FilePath rootFilePath = null;
+    private UsernamePasswordCredentialsImpl credentials=null;
+    private GitUsernamePasswordBind gitCredBind=null;
+
+    @Before
+    public void basicSetup() throws IOException {
+        //File init
+        rootDir = tempFolder.getRoot();
+        rootFilePath = new FilePath(rootDir.getAbsoluteFile());
+
+        //Credential init
+        credentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, "80y7g2fjekru8h9237ywes2", "GIt Username and Password Binding Test", this.username, this.password);
```

Comment:
Could you provide more context or a comment for the purpose of the magic string?  I assume that `80y7g2fjekru8h9237ywes2` is the credentials ID.  I'm accustomed to them having a different format.  A comment would help here to explain that this is a credentials ID.
```suggestion
        credentials = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, "80y7g2fjekru8h9237ywes2", "GIt Username and Password Binding Test", this.username, this.password);
```

Reply:
I will use a hash function for this purpose, it will be more certain than a magic string.